### PR TITLE
Create metapackage

### PR DESCRIPTION
**Pull request by RobPasMue**
Resolving #34 #13

### CODE REVIEW

**Comment by RobPasMue** [comment]
Project adapted to pyproject.toml.
Working on setting up CI/CD now.

**Comment by RobPasMue** [comment]
Current situation:
- I have cleaned up the repository in order to use a single ``pyproject.toml`` file with poetry (this allows us to remove the requirement files and let dependabot update the reqs regularly)
- We have created a "core" install and "extras" installs (for fluent-all, mapdl-all, all etc.)
- Vale has been implemented as well for the docs (even though it's only the main README.rst and a simple index.rst)
- Pre-commit implemented for style checks
- A new versioning for this package has been conceived (YYYY.X.dev0 for main where YYYY is the release year and X is the release within the year)
- A proper CI workflow has been implemented in which we are testing: code style, docs style, core build, "extras" build (nothing depends on it if it fails), packaging stage and release.

Future works:
- It looks like Python 3.10 is failing due to VTK in some of our PyAnsys libraries. I will need to look into that before merging so that everything works. Probably will have to open PRs in those libraries where Python 3.10 is not supported.
- We should also implement the automatic "milestone" generation. I already have some "local" proposal for this but I will have to handle it in a different PR (this one is already growing too much)


This being said, I'll leave this PR for review.

File: .github/workflows/ci-build.yml
Comment (jorgepiloto) [inline .github/workflows/ci-build.yml:35]:
```suggestion
        run: pre-commit run --all-files --show-diff-on-failure
```

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
This looks really good, @RobPasMue. Just a couple of things:

- No need to have a `_version.py` file. Let us have everything inside `__init__.py`. This allows also to simplify the very first lines in `conf.py`.

- I agree on using `poetry` for this. It would be nice to commit the `lock.file` too.

- Additionally to the wheel and source artifacts, we could ship the "wheelhouse", as PyMAPDL does, see [these lines](https://github.com/pyansys/pymapdl/blob/38d33037e2f780e9f694ec9abdf9db9aab837a6a/.github/workflows/ci.yml#L97-L111).

**Comment by RobPasMue** [comment]
> This looks really good, @RobPasMue. Just a couple of things:
> 
> * No need to have a `_version.py` file. Let us have everything inside `__init__.py`. This allows also to simplify the very first lines in `conf.py`.
> * I agree on using `poetry` for this. It would be nice to commit the `lock.file` too.
> * Additionally to the wheel and source artifacts, we could ship the "wheelhouse", as PyMAPDL does, see [these lines](https://github.com/pyansys/pymapdl/blob/38d33037e2f780e9f694ec9abdf9db9aab837a6a/.github/workflows/ci.yml#L97-L111).

Thank you for your comments @jorgepiloto I fully agree with all of them. Let me change them :)

**Comment by RobPasMue** [comment]
@jorgepiloto @akaszynski 

Problem with the dependencies appears when trying to create the lock file. We require resolution of this conflict before continuing. This is the conflict which exists between the "extra" dependencies.

image

**Comment by akaszynski** [comment]
> @jorgepiloto @akaszynski
> 
> Problem with the dependencies appears when trying to create the lock file. We require resolution of this conflict before continuing. This is the conflict which exists between the "extra" dependencies.
> 
> image

Open an issue for this on the corresponding DPF repos, but note that they're just getting a version that's working for 2022R2 and likely won't have a release that works for 2023R1 by 26 August. Pinging @PProfizi if you have anything to add here.

**Comment by RobPasMue** [comment]
Created new issues in this same repo for:
- [ ] Supporting Python 3.10 for all PyAnsys public libraries #36 
- [ ] Including ``ansys-fluent-visualization`` as a dependency #37 

Also issues in other repos have been created:
- [ ] pyansys/pydpf-post VTK dependency is old (<9.1.0) https://github.com/pyansys/pydpf-post/issues/107
- [ ] pyansys/pyfluent-visualization VTK dependency is too strict (only 9.1.0 is allowed) https://github.com/pyansys/pyfluent-visualization/issues/94

**Comment by akaszynski** [comment]
> Including `ansys-fluent-visualization` as a dependency [Including `ansys-fluent-visualization` as a dependency #37](https://github.com/pyansys/pyansys_landing/issues/37)

I think this is fine to be optional.

**Comment by RobPasMue** [comment]
> > Including `ansys-fluent-visualization` as a dependency [Including `ansys-fluent-visualization` as a dependency #37](https://github.com/pyansys/pyansys_landing/issues/37)
> 
> I think this is fine to be optional.

Yep, it should be optional, as part of the ``fluent-all`` and ``all`` targets. The thing is that this package clashes with the ``ansys-dpf-*`` packages right now. That's why I opened the issue to include it in the future once we resolve the conflicts.

File: pyproject.toml
Comment (MaxJPRey) [inline pyproject.toml:32]: It is not supported yet, correct?
```suggestion

```

File: pyproject.toml
Comment (MaxJPRey) [inline pyproject.toml:8]:
```suggestion
version = "2022.1.dev0"

```

File: pyproject.toml
Comment (RobPasMue) [reply pyproject.toml:32]: Agreed! At least not for now :)

File: pyproject.toml
Comment (RobPasMue) [reply pyproject.toml:8]: I would leave it as it is. The current "dev" version is intended to target 2023R1.

File: pyproject.toml
Comment (akaszynski) [inline pyproject.toml:51]: With https://github.com/pyansys/pymapdl-reader/pull/147 and https://github.com/pyansys/pymapdl-reader/actions/runs/2794314543, this should work now.

```suggestion
ansys-mapdl-reader = {version = "==0.51.15", optional = true}
```

File: pyproject.toml
Comment (akaszynski) [inline pyproject.toml:41]:
```suggestion
ansys-mapdl-core = "==0.63.1"

```

File: pyproject.toml
Comment (akaszynski) [inline pyproject.toml:43]: These will have to be updated. As it stands, `pydpf-core==0.5.0` is for 2022R2, which we're still waiting on.

File: pyproject.toml
Comment (akaszynski) [inline pyproject.toml:52]: We'll have to bring this up in a discussion tomorrow.

File: pyproject.toml
Comment (akaszynski) [inline pyproject.toml:55]: Why is Sphinx or the ansys-sphinx-theme required for this metapackage?

File: pyproject.toml
Comment (akaszynski) [reply pyproject.toml:8]: Agreed. Resolving...

File: pyproject.toml
Comment (akaszynski) [inline pyproject.toml:31]:
```suggestion
    'Programming Language :: Python :: 3.9',
    'Programming Language :: Python :: 3.10',
```

Comment (akaszynski) [inline pyproject.toml:31]:
```suggestion
    'Programming Language :: Python :: 3.9',
    'Programming Language :: Python :: 3.10',
```

Comment (akaszynski) [reply pyproject.toml:31]: I honestly have no idea why the text is all red.

Comment (RobPasMue) [reply pyproject.toml:31]: Sounds weird. I don't see anything in red. I'll look into it anyway

image

File: pyproject.toml
Comment (RobPasMue) [reply pyproject.toml:43]: At the moment, the latest available one is 0.4.2 (I refer to the released versions). As soon as they publish 0.5.0, dependabot will let us know

File: pyproject.toml
Comment (RobPasMue) [reply pyproject.toml:51]: Thanks @akaszynski for your work!

File: pyproject.toml
Comment (RobPasMue) [reply pyproject.toml:52]: Agreed. Planning on. There is an issue open as well. https://github.com/pyansys/pyfluent-visualization/issues/94

File: pyproject.toml
Comment (RobPasMue) [reply pyproject.toml:55]: They are optional, so they won't get installed by default. They are part of the ``docs`` target, which builds the PyAnsys Landing page. That's why it's needed.

**Comment by RobPasMue** [comment]
@jorgepiloto @MaxJPRey @akaszynski I think, as it is, we can merge the PR. Can you please review it (and approve) when possible? Remaining open points already have issues created in this same repo

**Comment by jorgepiloto** [comment]
Sure, let me do a quick review on this, @RobPasMue 👍🏽

File: doc/source/conf.py
Comment (jorgepiloto) [inline doc/source/conf.py:16]: These lines can be simplified by doing:

```python
import pyansys

release = version = pyansys.__verison__
```

File: pyproject.toml
Comment (jorgepiloto) [inline pyproject.toml:63]: I am not sure if it is possible to reference extra groups like this:

```toml
all = ["mapdl-all", "fluent-all"]
```

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Just one change request regarding the `conf.py` file and a tiny comment. This looks really good. This solution is really smart, congratulations @RobPasMue 👏🏽

File: pyproject.toml
Comment (RobPasMue) [reply pyproject.toml:63]: Yeah, that'd be great. But nope 😆 not  possible. Already tried it in the commit history. I wish that could be possible.

File: doc/source/conf.py
Comment (RobPasMue) [reply doc/source/conf.py:16]: Agreed! Let me add that :)

**Review by jorgepiloto — APPROVED** [review]

File: .github/workflows/ci-build.yml
Comment (MaxJPRey) [inline .github/workflows/ci-build.yml:63]:
```suggestion
```

File: .github/workflows/ci-build.yml
Comment (MaxJPRey) [inline .github/workflows/ci-build.yml:139]:
```suggestion
```

File: .github/workflows/ci-build.yml
Comment (MaxJPRey) [inline .github/workflows/ci-build.yml:281]: Do we want a dedicated token for the metapackage?
I can provide one.

**Comment by MaxJPRey** [comment]
Well done @RobPasMue. This is excellent.

**Review by MaxJPRey — APPROVED** [review]

File: .github/workflows/ci-build.yml
Comment (RobPasMue) [reply .github/workflows/ci-build.yml:281]: That sounds right. There is already a PyPi project for this https://pypi.org/project/pyansys/ Could you get access to it?

**Pull request merged**
